### ISSUE TITLE
How to install this on my raspberry pi?

### ISSUE THREAD

**Issue by harrydehix** [bytesnz/vproweather#4]
Hello,
I am the owner of a Vantage Pro 2 and I want to build a web API to my weather station using my Raspberry Pi and your program. The Vantage Pro 2 is connected to the Raspberry with the data logger via USB port. I know how to make web API's, that's not my problem, I just don't know how to make your program run on my Raspberry Pi. I am a Linux beginner and only know standard commands like cd, apt, pwd, ls, nano, .... I have hardly any idea about the C programming language. I know that I probably need to compile the program somehow, but I have little idea about it, unfortunately. I would be infinitely grateful if you could explain to me how, as a Linux beginner, I can get this program to run on my Raspberry Pi. I've been looking for a program like this for half an eternity and have never been this close to getting it....
With kind regards,
harrydehix

**Comment by bytesnz** [comment]
Hi harrydehix. It should be a case of (on your Pi):
1. ensuring GCC and make are installed (with `apt`)
2. cloning this repo (`git clone`)
3. going into the project directory
4. running `make`.

That should create the executable `vproweather` that can then run.
```shell
sudo apt install gcc make git
git clone https://github.com/bytesnz/vproweather
cd vproweather
make
./vproweather
```
If you have trouble, I have compiled it on my phone (currently without other technology for a couple of months) and this should work on a Pi, so can try and figure out how to upload here.

Let me know how you get on.

**Comment by bytesnz** [comment]
What were you planning on writing the API in?

**Comment by harrydehix** [comment]
Javascript (NodeJS). With NodeJS I can easily call shell commands using exec("command"), await the result and use it in my application.

**Comment by harrydehix** [comment]
Oh wow, I overlooked your first comment😂. Thank you for your detailed explanation, I'm going to try it out now!

**Comment by bytesnz** [comment]
> Javascript (NodeJS). With NodeJS I can easily call shell commands using exec("command"), await the result and use it in my application.

Nice. Would be interested in seeing it if you publish it.

Let me know how you get on.

**Comment by harrydehix** [comment]
After I get your program to work, I'm going to create a public repo😄. It's going to be a REST API.

Okay I successfully installed your program! And it works!! ~~But the only thing is... how do I know which serial device is my weather station? Is there a command to see all connected serial devices?~~
Okay thanks to weeex (I used that tool before) I got the "device url".😄 I'll start coding now and comment here after I finished my api! Thanks a lot man, never thought it could be that easy😄
